server/controllers/settings.js

```diff
@@ -5,6 +5,28 @@ import {ADMIN_ROLE, volunteerRoles} from '../../common/constants'
 import config from '../config'
 import Settings from '../models/settings'
 
+import nodeGeocoder from 'node-geocoder'
+import {fieldTypes} from '../../common/constants'
+import {getFieldsByType} from '../lib/questionnaire-helpers'
+import locationSchema from '../models/location-schema'
+
+// Initialize geocoder options for pre save method
+const geocoder = nodeGeocoder({
+  provider: 'google',
+  formatter: null
+})
+
+/**
+ * Hook a method to construct the geolocation of the address
+ */
+async function updateLocationSettings(address, id){
+  const [result] = await geocoder.geocode(address)
+  if (result && result != undefined && result != null && result != {}) {
```

Comment:
You need something like this to show the error on the form:
```js
if (!result) return next(new ValidationError({x: y}))
```
where x is the field of the model that has the error and y is the error message. Then when there's no geocode result the form will show field x in red with the error message y.